salesforce/php/register-back.php

```diff
@@ -0,0 +1,304 @@
+    if(!$responseGood){
+        log_entry("failed to create user");
+        $subject = $exception->getMessage();
+        log_entry($subject);
+        mail('web-development@bethel.edu', $subject, $subject, "From: $from\n");
+        return null;
+    }
+    $output = print_r($createResponse,1);
+    log_entry('create user : ' . $output);
+    //Creates response
+    $errorMessage = json_decode(json_encode($createResponse[0]), true);
+    $user_id = $createResponse[0]->id;
+    add_permission_set($user_id);
+    return $user_id;
+
+}
+
+function add_permission_set($user_id){
+    global $mySforceConnection;
+    global $PERMISSIONSETID;
+    // give permission set
+    $sObject = new stdclass();
+    $sObject->AssigneeId = $user_id;
+    $sObject->PermissionSetId = $PERMISSIONSETID;
+    return $mySforceConnection->create(array($sObject), 'PermissionSetAssignment');
+}
+
+//Setting Variables, as well as declaring the environment
+$staging = strstr(getcwd(), "/staging");
+$mail_to = "web-development@bethel.edu";
+$mail_from = "salesforce-register@bethel.edu";
+$subject = "salesforce register submission";
+$message = "";
+
+//prepare a URL for returing
+if ($staging){
+    $url = 'https://staging.bethel.edu/admissions/apply/';
+}else{
+    $url = 'https://www.bethel.edu/admissions/apply';
+}
+
+$email = $_POST["email"];
+$email = strtolower($email);
+$email = trim($email);
+
+log_entry($email);
+$first = $_POST["first"];
+$last = $_POST["last"];
+
+$search_email = escapeEmail($email);
+$search_email = '{' . $search_email . '}';
+
+function create_interaction($first, $last, $email){
+    global $mySforceConnection;
+
+    $utm_source = '';
+    $utm_medium = '';
+    $utm_content = '';
+    $utm_campaign = '';
+
+    if( $_COOKIE['utm_source'] )
+        $utm_source = $_COOKIE['utm_source'];
+    if( $_COOKIE['utm_medium'] )
+        $utm_medium = $_COOKIE['utm_medium'];
+    if( $_COOKIE['utm_content'] )
+        $utm_content = $_COOKIE['utm_content'];
+    if( $_COOKIE['utm_campaign'] )
+        $utm_campaign = $_COOKIE['utm_campaign'];
+
+
+    $sObject = new stdclass();
+    $sObject->First_Name__c = $first;
+    $sObject->Last_Name__c = $last;
+    $sObject->Email__c = $email;
+    $sObject->Interaction_Source__c = 'Account Register';
+    $sObject->Lead_Source__c = 'Website';
+    $sObject->Source_Detail__c = 'Application';
+    $sObject->Marketing_Campaign__c = $utm_campaign;
+    $sObject->Marketing_Source__c = ucwords(str_replace('_', ' ', $utm_source));
+    $sObject->Marketing_Medium__c = ucwords(str_replace('_', ' ', $utm_medium));
+
+    try{
+        $createResponse = $mySforceConnection->create(array($sObject), 'Interaction__c');
+    }catch(Exception $e){
+        log_entry("failed to create interaction");
+        $subject = $e->getMessage();
+        log_entry($subject);
+        // todo update before golive
+        mail('e-jameson@bethel.edu',$subject,$subject,"From: $from\n");
```

Comment:
just a reminder to update this entry before this gets merged

Reply:
ah, I need to delete the register-back.php file. Its just there for easy reference.